src/daft-core/src/array/ops/hll_merge.rs

```diff
@@ -0,0 +1,43 @@
+use common_error::DaftResult;
+
+use crate::{
+    array::ops::{as_arrow::AsArrow, DaftHllMergeAggable},
+    datatypes::{FixedSizeBinaryArray, UInt64Array},
+};
+use hyperloglog::HyperLogLog;
+
+use crate::array::ops::GroupIndices;
+
+impl DaftHllMergeAggable for FixedSizeBinaryArray {
+    type Output = DaftResult<UInt64Array>;
+
+    fn hll_merge(&self) -> Self::Output {
+        let mut final_hll = HyperLogLog::default();
+        for byte_slice in self.as_arrow().values_iter() {
```

Comment:
we should only be iterating over valid values. Let's use `.iter()` and skip the Nulls?

Reply:
Updated!